internal/component/otelcol/config_datadog.go

```diff
@@ -0,0 +1,221 @@
+}
+
+// Convert converts args into the upstream type.
+func (args *DatadogHostMetadataArguments) Convert() *datadogexporter.HostMetadataConfig {
+	if args == nil {
+		return nil
+	}
+
+	return &datadogexporter.HostMetadataConfig{
+		Enabled:        args.Enabled,
+		HostnameSource: args.HostnameSource,
+		Tags:           args.Tags,
+	}
+}
+
+func (args *DatadogHostMetadataArguments) SetToDefault() {
+	*args = DatadogHostMetadataArguments{
+		Enabled:        true,
+		HostnameSource: datadogexporter.HostnameSourceConfigOrSystem,
+	}
+}
+
+// TracesConfig holds the configuration settings for the Datadog trace exporter
+// See https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter#TracesConfig for more
+type DatadogTracesArguments struct {
+	Endpoint                  string            `alloy:"endpoint,attr,optional"`
+	IgnoreResources           []string          `alloy:"ignore_resources,attr,optional"`
+	SpanNameRemappings        map[string]string `alloy:"span_name_remappings,attr,optional"`
+	SpanNameAsResourceName    bool              `alloy:"span_name_as_resource_name,attr,optional"`
+	ComputeStatsBySpanKind    bool              `alloy:"compute_stats_by_span_kind,attr,optional"`
+	ComputeTopLevelBySpanKind bool              `alloy:"compute_top_level_by_span_kind,attr,optional"`
+	PeerTagsAggregation       bool              `alloy:"peer_tags_aggregation,attr,optional"`
+	PeerTags                  []string          `alloy:"peer_tags,attr,optional"`
+	TraceBuffer               int               `alloy:"trace_buffer,attr,optional"`
+}
+
+func (args *DatadogTracesArguments) Convert() *datadogexporter.TracesConfig {
+	if args == nil {
+		return nil
+	}
+
+	return &datadogexporter.TracesConfig{
+		TCPAddrConfig:             confignet.TCPAddrConfig{Endpoint: args.Endpoint},
+		IgnoreResources:           args.IgnoreResources,
+		SpanNameRemappings:        args.SpanNameRemappings,
+		SpanNameAsResourceName:    args.SpanNameAsResourceName,
+		ComputeStatsBySpanKind:    args.ComputeStatsBySpanKind,
+		ComputeTopLevelBySpanKind: args.ComputeTopLevelBySpanKind,
+		PeerTagsAggregation:       args.PeerTagsAggregation,
+		PeerTags:                  args.PeerTags,
+		TraceBuffer:               args.TraceBuffer,
+	}
+}
+
+func (args *DatadogTracesArguments) SetToDefault() {
+	*args = DatadogTracesArguments{
+		IgnoreResources: []string{},
+	}
+}
+
+// MetricsExporterConfig holds the configuration settings for the Datadog metrics exporter
+type DatadogMetricsArguments struct {
+	DeltaTTL                           int64                     `alloy:"delta_ttl,attr,optional"`
+	Endpoint                           string                    `alloy:"endpoint,attr,optional"`
+	ResourceAttributesAsTags           bool                      `alloy:"resource_attributes_as_tags,attr,optional"`
+	InstrumentationScopeMetadataAsTags bool                      `alloy:"instrumentation_scope_metadata_as_tags,attr,optional"`
```

Comment:
nit: I would suggest to use a block to wrap the two attributes above because in otel to follow the pattern in otel. Following upstream's structure makes it easier to detect changes and might be necessary if one day we invest in making tools to ease our dependency updates.

Reply:
Sounds good, modified to use the MetricsExporterConfig in the struct